### ISSUE TITLE
Equivalent to `lsadump::setntlm /user:USERNAME/ntlm:HASH` for Kerberos keys

### ISSUE THREAD

**Issue by gettalong** [gentilkiwi/mimikatz#145]
## Tl;dr

Is it possible to set the Kerberos keys (e.g. an AES256-CTS-HMAC-SHA1-96 key) for a user?

## Longer version

Maybe a bit crazy but I intend to migrate user *with* their passwords from an old AD to a new one. I can easily retrieve the NTLM hashes and the kerberos keys using the [impacket tools](https://github.com/CoreSecurity/impacket/). And then the `lsadump::setntlm` method comes in quite handy to set the passwords via the the NTLM hash for each user (we have another script for generating the basic user account objects beforehand). However, I didn't find an equivalent method for changing the Kerberos keys.

Since NTLM hashes are only used during rc4-hmac Kerberos authentication, the stronger authentication types won't be available. Which means that we wouldn't be able to turn off support for rc4-hmac.

So is there some way in Mimikatz (or maybe other tools) to set the various Kerberos keys for a user account?

**Comment by vletoux** [comment]
Short answer: 
no with setnlm
yes with DCShadow
Beware that the hash includes the domain name and if you are planing to change it this will be a problem.
Did you check also a password filter ?

**Comment by gettalong** [comment]
Thanks for your quick reply!

Your comment makes it clear that transferring the Kerberos keys won't be possible since we indeed change the domain name. What we will probably do is making the transition to the new AD first and in a second step give the users some months to change/set their passwords so that we can disable RC4-HMAC later.
